solr/solrj/src/java/org/apache/solr/client/solrj/impl/HttpSolrClientBuilderBase.java

```diff
@@ -47,6 +47,37 @@ public abstract class HttpSolrClientBuilderBase<
 
   public abstract C build();
 
+  /**
+   * Provide a seed HttpSolrClient for the builder values, values can still be overridden by using
+   * builder methods
```

Comment:
Should document that any state of the current builder that has been set will not be changed.
Although... it'd be intuitive & simpler for this to work the reverse IMO.  Basically, withHttpClient would set whatever state it sets no matter if withWhatever has been called already.
Even simpler and less error-prone would be to only have this be a constructor option.